File: flutter app/app/lib/Controllers/UserController.dart
import 'package:get/get.dart';
import 'package:usm_mobile/Models/user_model.dart';
import 'package:usm_mobile/Repositories/UserRepo.dart';

class UserController extends GetxController {
  static UserController _userController;
  static UserController getInstance() {
    if (_userController == null)
      return _userController = UserController();
    else
      return _userController;
  }

  UserModel userinfo = UserModel();
  var userRepo = UserRepo();
  @override
  onInit() {
    super.onInit();
  }

  void _fetchData() async {
    userinfo = await userRepo.register();
    print(userinfo.toJson());
  }

  Future<bool> signinControl(String email, String password) async {
    userinfo = await userRepo.signin(email, password);
    if (userinfo.id == -1 || userinfo == null)
      return false;
    else {
      return true;
    }
  }
}

File: desktop-app/app/src/main/java/app/Controllers/UserController.java
package app.Controllers;

import app.Models.UserModel;
import app.repository.UserRepository;

public class UserController {
    UserRepository userRepository;
    private UserModel userModel;

    private static UserController userController;
    private UserController(){
        init();
    }
    private void init(){
        userRepository = new UserRepository();
        userModel = new UserModel();
    }
    public static UserController getInstance(){
        if(userController == null)
        {
            userController = new UserController();
            return userController;
        }else
        return userController;
    }
    
    public boolean checkSignIn(String email, String password)
    {
        userModel = userRepository.signIn(email, password);
        if(userModel.getId() == -1 ) return false;
        else return true;
        
    } 
    
    
    public UserModel getUserModel() {
        return this.userModel;
    }
    
    public void setUserModel(UserModel userModel) {
        this.userModel = userModel;
    }
}
